src/main/drivers/bus.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "drivers/bus_i2c.h"
+#include "drivers/bus_spi.h"
+
+typedef enum {
+    BUSTYPE_NONE = 0,
+    BUSTYPE_I2C  = 1,
+    BUSTYPE_SPI  = 2
+} busType_e;
+
+/* Ultimately all hardware descriptors will go to target definition files. 
+ * Driver code will merely query for it's HW descriptor and initialize it */
+typedef enum {
+    DEVHW_NONE,
+    DEVHW_MAX7456,
+    DEVHW_BMP280
+} devHardwareType_e;
+
+typedef struct busDeviceDescriptor_s {
+    busType_e           busType;
+    devHardwareType_e   devHwType;
+    union {
+        struct {
+            SPIDevice   spiBus;
+            ioTag_t     csnPin;
+        } spi;
+        struct {
+            I2CDevice   i2cBus;
+            uint8_t     address;
+        } i2c;
+    } dev;
+} busDeviceDescriptor_t;
+
+/*
+typedef struct busDevice_s {
+    busType_e           busType;
+    union {
+        struct {
+            SPIDevice   spiBus;
+            IO_t        csnPin;
+        } spi;
+        struct {
+            I2CDevice   i2cBus;
+            uint8_t     address;
+        } i2c;
+    } busdev;
+} busDevice_t;
+*/
+
+typedef union busDevice_u {
+    struct deviceSpi_s {
+        IO_t csnPin;
+    } spi;
+    struct deviceI2C_s {
+        uint8_t address;
+    } i2c;
+} busDevice_t;
+
+#ifdef __APPLE__
```

Comment:
Why is this needed? __APPLE__ should only be defined when compiling for macOS and iOS and I guess we're not supporting any of those targets anytime soon :-)

Reply:
True, but it's better to keep the source compilable :smile:

Reply:
No problem with that being left in the PR. It just grabbed my attention cause I wasn't really expecting a test for Apple's platform on code running on MCUs!

Now that I think about it, ßF added support for running on a simulator and we eventually should do that too. Not sure how that simulator works but it might need that special handling on macOS.

Reply:
Actually, I was mostly concerned with chip-select lines left floating for not (yet) initialized SPI devices. Might be the cause of #2074.